### PR TITLE
Feat: sitemap integration filter

### PR DESCRIPTION
**Pull request by bholmesdev**
## Changes

- Add `filter` sitemap integration option
- Add `canonicalURL` sitemap integration option
- update sitemap integration README

## Testing

Post-launch hopefully 😉 

## Docs

See new integration README

### CODE REVIEW

File: packages/integrations/sitemap/src/index.ts [hunk 1]
```diff
@@ -25,11 +47,15 @@ export default function createPlugin(): AstroIntegration {
 				config = _config;
 			},
 			'astro:build:done': async ({ pages, dir }) => {
-				const finalSiteUrl = config.site;
+				const finalSiteUrl = canonicalURL || config.site;
 				if (!finalSiteUrl) {
+					console.warn('The Sitemap integration requires either the `site` astro.config option or `canonicalURL` integration option. Skipping.');
```
Comment (bholmesdev) [inline packages/integrations/sitemap/src/index.ts:52]: We don't have access to the `logger` from integrations. Happy to remove this if we don't want `console`!

**Review by natemoo-re — APPROVED** [review]
LGTM! Nice work!

File: packages/integrations/sitemap/src/index.ts (hunk 1 above)
Comment (natemoo-re) [reply packages/integrations/sitemap/src/index.ts:52]: This is definitely something to tackle as we improve these APIs. I think `console` is fine for now!

Comment (bholmesdev) [reply packages/integrations/sitemap/src/index.ts:52]: Yeah, maybe we could expose Astro's `logger` from the `config:setup` hook?

**Pull request merged**